packages/core/src/modules/credentials/messages/IssueCredentialMessage.ts

```diff
@@ -48,14 +47,9 @@ export class IssueCredentialMessage extends AgentMessage {
   public get indyCredential(): Cred | null {
     const attachment = this.credentialAttachments.find((attachment) => attachment.id === INDY_CREDENTIAL_ATTACHMENT_ID)
 
-    // Return null if attachment is not found
-    if (!attachment?.data?.base64) {
-      return null
-    }
-
     // Extract credential from attachment
-    const credentialJson = JsonEncoder.fromBase64(attachment.data.base64)
+    const credentialJson = attachment?.data?.getDataAsJson()
 
-    return credentialJson
+    return credentialJson as Cred
```

Comment:
This casting removes the `null`. Is this intentional? If so, could you remove it from the function signature aswell?

Reply:
hmm... yeah intention is that according the linter type of credentialJson is unknown if I leave out `as Cred` which 'not assignable to Cred'

Reply:
basically that's why i had sth along the lines of  
```typescript
const credentialJson = attachment?.data ? attachment.data.getDataAsJson() : null 
```
in the first commit... Again greatful for any help

Reply:
If we use optional chaining, `.?` we short circuit to return an `undefined`. This means that the value we want to return from the function could be undefined. 

This could happen if `attachment` or `attachment.data` does not exist. 

So if we want a 100% to return an object and not `null` from this function we must check if these two fields exist:

```ts
If (attachment?.data) {
  // some case handling
}
```

After this we 100% have `attachment.data ` to call `getDataAsJson()` on. 

If we do not want to return the object OR `null`, as the api is right now, we could go back to your ternary expression.

Reply:
or we use try...catch?